swig/cpp/src/Libyang.hpp

```diff
@@ -47,48 +45,48 @@ class Error
 /* add custom deleter for Context class */
 public:
     Error() {
-        _err = ly_errno;
-        _vecode = ly_vecode;
-        _errmsg = ly_errmsg();
-        _errpath = ly_errpath();
-        _errapptag = ly_errapptag();
+        libyang_err = ly_errno;
+        libyang_vecode = ly_vecode;
+        libyang_errmsg = ly_errmsg();
+        libyang_errpath = ly_errpath();
+        libyang_errapptag = ly_errapptag();
     };
     ~Error() {};
-    LY_ERR err() throw() {return _err;};
-    LY_VECODE vecode() throw() {return _vecode;};
-    const char *errmsg() const throw() {return _errmsg;};
-    const char *errpath() const throw() {return _errpath;};
-    const char *errapptag() const throw() {return _errapptag;};
+    LY_ERR err() throw() {return libyang_err;};
+    LY_VECODE vecode() throw() {return libyang_vecode;};
+    const char *errmsg() const throw() {return libyang_errmsg;};
+    const char *errpath() const throw() {return libyang_errpath;};
+    const char *errapptag() const throw() {return libyang_errapptag;};
 private:
-    LY_ERR _err;
-    LY_VECODE _vecode;
-    const char *_errmsg;
-    const char *_errpath;
-    const char *_errapptag;
+    LY_ERR libyang_err;
+    LY_VECODE libyang_vecode;
+    const char *libyang_errmsg;
+    const char *libyang_errpath;
+    const char *libyang_errapptag;
 };
 
 class Context
 {
 public:
     Context(struct ly_ctx *ctx, S_Deleter deleter);
-    Context(const char *search_dir = NULL, int options = 0);
+    Context(const char *search_dir = nullptr, int options = 0);
```

Comment:
This one should probably be marked as `explicit`, so that there's no automatic cast from string literals to a `Context` instance.

Same in other classes.

Reply:
Do you mean `explicit Context(const char *search_dir = nullptr, int options = 0);` ?

Reply:
Yes